backend/app/services/implementations/onboarding_request_service.py

```diff
@@ -38,3 +38,32 @@ def create_onboarding_request(self, userInfo):
             raise e
 
         return new_onboarding_request.to_serializable_dict()
+
+    def get_all_onboarding_requests(self):
+        onboarding_request_objects = []
+
+        try:
+            for request in OnboardingRequest.objects:
```

Comment:
Instead of using the objects directly, it would be best to use the built-in Mongo filters! That way, we can get potential performance increases (AFAIK, the MongoDB server is written in C++) and reduce the complexity of the filtering logic needed. Doing so would also eliminate the need to iterate the results at all here.

You can refer to the docs [here](https://docs.mongoengine.org/guide/querying.html#filtering-queries) (in particular, see the segment on embedded documents).

Reply:
Sounds good! Makes sense :)